src/rust/src/pkcs7.rs

```diff
@@ -164,6 +166,273 @@ fn encrypt_and_serialize<'p>(
+                        )),
+                    ));
+                }
+            };
+
+            // Decrypt the content using the key and proper algorithm
+            let encrypted_content = match enveloped_data.encrypted_content_info.encrypted_content {
+                Some(content) => content,
+                None => {
+                    return Err(CryptographyError::from(
+                        pyo3::exceptions::PyValueError::new_err(
+                            "The EnvelopedData structure does not contain encrypted content.",
+                        ),
+                    ));
+                }
+            };
+            let decrypted_content = symmetric_decrypt(py, algorithm, mode, encrypted_content)?;
+            pyo3::types::PyBytes::new(py, decrypted_content.as_slice())
+        }
+        _ => {
+            return Err(CryptographyError::from(
+                pyo3::exceptions::PyValueError::new_err(
+                    "The PKCS7 data is not an EnvelopedData structure.",
+                ),
+            ));
+        }
+    };
+
+    // If text_mode, strip the header if possible, else return an error
+    let plain_data = plain_content.as_bytes();
+    let plain_data = if options.contains(types::PKCS7_TEXT.get(py)?)? {
+        match plain_data.strip_prefix(b"Content-Type: text/plain\r\n\r\n") {
+            Some(stripped_data) => stripped_data,
+            None => {
+                return Err(CryptographyError::from(
+                    pyo3::exceptions::PyValueError::new_err(
+                        "No 'Content-Type' header to be deleted in the decrypted data. Please remove the 'text' option.",
+                    ),
+                ));
+            }
+        }
+    } else {
+        plain_data
+    };
+
+    let plain_data = pyo3::types::PyBytes::new(py, plain_data);
+    Ok(plain_data)
+}
+
+fn check_decrypt_parameters<'p>(
+    py: pyo3::Python<'p>,
+    certificate: &pyo3::Bound<'p, x509::certificate::Certificate>,
+    private_key: &pyo3::Bound<'p, pyo3::PyAny>,
+    options: &pyo3::Bound<'p, pyo3::types::PyList>,
+) -> Result<(), CryptographyError> {
+    // Check if RSA encryption with PKCS1 v1.5 padding is supported (dependent of FIPS mode)
+    if cryptography_openssl::fips::is_enabled() {
+        return Err(CryptographyError::from(
+            exceptions::UnsupportedAlgorithm::new_err((
+                "RSA with PKCS1 v1.5 padding is not supported by this version of OpenSSL.",
+                exceptions::Reasons::UNSUPPORTED_PADDING,
+            )),
+        ));
+    }
+
+    // Check if all options are from the PKCS7Options enum
+    let pkcs7_option_type = types::PKCS7_TEXT.get(py)?.get_type();
+    if !options
+        .iter()
+        .all(|opt| opt.is_instance(&pkcs7_option_type).unwrap())
+    {
+        return Err(CryptographyError::from(
+            pyo3::exceptions::PyValueError::new_err("options must be from the PKCS7Options enum"),
+        ));
+    }
```

Comment:
And in this case you can unroll the loop:

```
for opt in options.iter() {
    if !opt.is_instance(&pkcs7_option_type)? {
        return Err(...);
    }
}

Reply:
Thanks! I'll go with this